src/ir_passes/assume.rs

```diff
@@ -0,0 +1,164 @@
+use crate::{
+    ast::{self, FnAssume},
+    ir::{self, Ctx, ExprIdx, PropIdx},
+    ir_visitor::{Action, Visitor},
+    utils::Binding,
+};
+
+/// Generates default assumptions to the Filament program for assumptions using custom functions
+#[derive(Default)]
+pub struct Assume;
+
+impl ir::Component {
+    /// Interns new [ir::Expr]s generated from an [ast::Expr] given a binding on [ir::ExprIdx]
+    fn add_expr(
+        &mut self,
+        expr: ast::Expr,
```

Comment:
We shouldn't be working with `ast` in this directory. Once things are in the `ir`, we should only construct `ir` objects.